pkg/apis/pipeline/v1beta1/version_validation.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"knative.dev/pkg/apis"
+)
+
+// ValidateEnabledAPIFields checks that the enable-api-fields feature gate is set
+// to the wantVersion value and, if not, returns an error stating which feature
+// is dependent on the version and what the current version actually is.
+func ValidateEnabledAPIFields(ctx context.Context, featureName, wantVersion string) *apis.FieldError {
```

Comment:
@sbwsg I am trying to understand the usage of this function, its only referenced in the unit tests for now. How do we envision this being utilized in future?

Reply:
Ah, good catch, there's an example of usage over in the Step & Sidecar Workspaces PR: https://github.com/tektoncd/pipeline/pull/3700/files#diff-988b00fa4f188a96ff5901c0bbc9f65d4a8355417185d4fe0ae97d8c14e04417R131-R133

The below snippet ensures that the new field is only valid to use when `enable-api-fields` is `alpha`:

```go
	for stepIdx, step := range steps {
		if len(step.Workspaces) != 0 {
			errs = errs.Also(ValidateEnabledAPIFields(ctx, "step workspaces", config.AlphaAPIFields).ViaIndex(stepIdx).ViaField("steps"))
		}
```